docs/06-concepts/11-authentication/04-providers/06-custom-providers.md

```diff
@@ -113,25 +113,53 @@ class MyAuthenticationEndpoint extends Endpoint {
 
 The example above shows how to create an `AuthenticationResponse` with the auth token and user information.
 
-### Remove auth token
+## Managing Authentication Keys and Signing Out Users
 
-Signing out a user on all devices is made simple with the `signOutUser` method in the `UserAuthentication` class. The method removes all auth tokens associated with the user.
+Serverpod provides built-in methods for managing user authentication across multiple devices. Below are the key functions for signing out users and revoking authentication keys. It is crucial to use these methods to ensure that all required steps, such as notifying clients and clearing session data, are properly handled.
+
+### Revoking Authentication Keys (Per Device)
+
+To sign a user out from a specific device, you can revoke the authentication key associated with that session using the `revokeAuthKey` method. This method targets a specific session and allows the user to remain signed in on other devices.
+
+#### Usage:
 
 ```dart
-class AuthenticatedEndpoint extends Endpoint {
-  @override
-  bool get requireLogin => true;
-  Future<void> logout(Session session) async {
-    await UserAuthentication.signOutUser(session);
-  }
+await UserAuthentication.revokeAuthKey(
+  session,
+  authKeyId: 'auth-key-id-here',
+);
+```
+
+#### Example: Fetching and Revoking an Authentication Key
+
+To revoke a specific authentication key, you first need to fetch the keys associated with a user. You can then select the key you want to revoke and call the `revokeAuthKey` method.
+
+```dart
+// Fetch all authentication keys for the user
+var authKeys = await AuthKey.db.find(
+  session,
+  where: (row) => row.userId.equals(userId),
+);
+
+// Revoke a specific key (for example, the last one)
+if (authKeys.isNotEmpty) {
+  var authKeyId = authKeys.last.id.toString();  // Convert the ID to string
+  await UserAuthentication.revokeAuthKey(
+    session,
+    authKeyId: authKeyId,
+  );
 }
 ```
 
-In the above example, the `logout` endpoint removes all auth tokens associated with the user. The user is then signed out and loses access to any protected endpoints.
+#### Importance of Using Built-In Methods
+
+Directly deleting authentication keys from the `AuthKey` table is not recommended, as this bypasses essential steps such as notifying connected clients and clearing the session's authentication data. The `revokeAuthKey` method performs several additional actions, including:
```

Comment:
Suggestion: We don't notify connected clients. We close all streaming connections and notify other servers that the key has been revoked.